src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -553,9 +558,10 @@ private PollingResult compareRemoteRevisionWithImpl(Job<?, ?> project, Launcher
             return BUILD_NOW;
         }
 
-        final BuildData buildData = fixNull(getBuildData(lastBuild));
-        if (buildData.lastBuild != null) {
-            listener.getLogger().println("[poll] Last Built Revision: " + buildData.lastBuild.revision);
+        BuiltRevisionMap builtRevisions = BuiltRevisionMap.forProject(project);
```

Comment:
Yes, real Actions are stored, Transient in-memory. But transient can be persisted.
AbstractProject has method   `  protected List<Action> createTransientActions() {` that should provide transient action through job api... But as i see it used only in matrix project and this weird, why not provide API for getting TransientAction from job?